src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -393,27 +394,125 @@ private Task StartAsyncCore(TransferFormat transferFormat)
-                }
-                catch (Exception ex)
+                using (var reader = new JsonTextReader(new StreamReader(await response.Content.ReadAsStreamAsync())))
                 {
-                    throw new FormatException("Invalid negotiation response received.", ex);
+                    JsonUtils.CheckRead(reader);
+                    JsonUtils.EnsureObjectStart(reader);
+
+                    string connectionId = null;
+                    List<AvailableTransport> availableTransports = null;
+
+                    var completed = false;
+                    do
+                    {
+                        switch (reader.TokenType)
+                        {
+                            case JsonToken.PropertyName:
+                                string memberName = reader.Value.ToString();
+
+                                switch (memberName)
+                                {
+                                    case "connectionId":
+                                        connectionId = JsonUtils.ReadAsString(reader, "connectionId");
+                                        break;
+                                    case "availableTransports":
+                                        JsonUtils.CheckRead(reader);
+                                        JsonUtils.EnsureArrayStart(reader);
+
+                                        availableTransports = new List<AvailableTransport>();
+                                        while (JsonUtils.CheckRead(reader))
+                                        {
+                                            if (reader.TokenType == JsonToken.StartObject)
+                                            {
+                                                availableTransports.Add(ParseAvailableTransport(reader));
+                                            }
+                                            else if (reader.TokenType == JsonToken.EndArray)
+                                            {
+                                                break;
+                                            }
+                                        }
+                                        break;
+                                }
+                                break;
+                            case JsonToken.EndObject:
+                                completed = true;
+                                break;
+                        }
+                    } while (!completed && JsonUtils.CheckRead(reader));
+
+                    return new NegotiationResponse
+                    {
+                        ConnectionId = connectionId,
+                        AvailableTransports = availableTransports
+                    };
                 }
             }
+            catch (Exception ex)
+            {
+                throw new FormatException("Invalid negotiation response received.", ex);
```

Comment:
`InvalidDataException` would be more appropriate (I know `FormatException` was in the old code as well, but...). `FormatException` is generally for `string.Format` and friends